### PR TITLE
Premium Content Block: User can't access premium content on renewal date.

### LINKED ISSUE #46242: Premium Content: Subscribers cannot see content on renewal date
According to PCYsg-4rX-p2, monthly and annual subscriptions that grant access to see premium content are renewed the day of expiry. I'm assuming they can happen at any time during the expiry day. However, the Premium Content block is not showing the protected content as soon as [the expiry date is reached](https://github.com/Automattic/wp-calypso/blob/e62f23a085b11a5279cbd6d59f1db5fce423dcff/apps/editing-toolkit/editing-toolkit-plugin/premium-content/subscription-service/class-token-subscription-service.php#L212-L215). In other words, between the expiry date at 00:00 UTC and when our async job renews the subscription, the plan is considered invalid and therefore the premium content is not visible to the subscriber.

A possible workaround is to add an additional day of grace to the expiry date check above, so the content is always visible regardless when the renewal happens:

```
if ( ( $end + DAY_IN_SECONDS ) > time() ) {
  return true;
}
```

Originally reported by @gwensmithx in p1602025594051900-slack-CDA5HLRD2.

### PR DESCRIPTION
**Pull request by rcrdortiz**
Fixes [#46242](https://github.com/Automattic/wp-calypso/issues/46242)

#### Changes proposed in this Pull Request:

* Since we can't guarantee that the subscription renewal will happen exactly at the time of the expiry of the subscription, we now add a one-day grace period to the check that determines if the user can or can't view the premium content. 

#### Testing instructions:

* Open a post that has premium content.
* Create a token with a subscription that expires today.
* Append the token to the URL.
* You should see the content.

#### Test evidence:

https://user-images.githubusercontent.com/1989914/153202351-56337c4d-425e-4140-b039-33d7dfc49562.mp4

### CODE REVIEW

**Review by dsas — COMMENTED** [review]
The code makes sense and I'm sure it works, I have low confidence I'm testing it properly though. Maybe more explicit written instructions would help, possibly some of them could be genericised for P2/fieldguide?

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php [hunk 1]
```diff
@@ -218,7 +218,7 @@ protected function validate_subscriptions( $valid_plan_ids, $token_subscriptions
 		foreach ( $token_subscriptions as $product_id => $token_subscription ) {
 			if ( in_array( $product_id, $product_ids, true ) ) {
 				$end = is_int( $token_subscription->end_date ) ? $token_subscription->end_date : strtotime( $token_subscription->end_date );
-				if ( $end > time() ) {
+				if ( ( $end + 24 * 3600 ) > time() ) {
```
[the remark this answers is not in the document]
Comment (BogdanUngureanu) [reply projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:221]: > , I did wonder about rounding up to the end of the day

I thought about it too but I think there's a chance of having the problem again in some cases (e.g. different timezones). Working with dates is hard, haha. :D

[the remark this answers is not in the document]
Comment (rcrdortiz) [reply projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:221]: The end_date of the subscription is stored in UTC, it should be independent of timezones.

Comment (dsas) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:221]: Oh, just FYI there is a WordPress constant called DAY_IN_SECONDS but for myself I'm pretty sure every dev knows how many seconds are in a day or hour...

[the remark this answers is not in the document]
Comment (dsas) [reply projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:221]: paYJgx-1Mt-p2

**Review by dsas — APPROVED** [review]
This all tests well, approving pending input from the discussion

**Comment by Copons** [comment]
Waiting on https://github.com/Automattic/wp-calypso/issues/52287 to be closed

**Comment by jeherve** [comment]
Closing this for now because of the lack of activity on this. We can always reopen in the future if needed.

**Pull request closed without merging**